engines/alg/game_spacepirates.cpp

```diff
@@ -0,0 +1,1955 @@
+	_shot_color = inSaveFile->readByte();
+	_shot_direction = inSaveFile->readByte();
+	_picked_start_splitter = inSaveFile->readUint16LE();
+	_current_world = inSaveFile->readSByte();
+	_selected_a_world = inSaveFile->readByte();
+	_selected_world_start = inSaveFile->readUint16LE();
+	_crystal_state = inSaveFile->readByte();
+	_shot_grin_reaper_count = inSaveFile->readByte();
+	_crystals_shot = inSaveFile->readByte();
+	_last_extra_life_score = inSaveFile->readUint32LE();
+	delete inSaveFile;
+	_game_loaded = true;
+	_gameInProgress = true;
+	_lives_loaded = _lives;
+	_shots_loaded = _shots;
+	_score_loaded = _score;
+	_difficulty_loaded = _difficulty;
+	_cur_scene = Common::String::format("scene%d", sceneNum);
+	_ChangeDifficulty(_difficulty);
+	if (!(_world_done[0] && _world_done[1] && _world_done[2] && _world_done[3]) && _current_world != -1) {
+		_cur_scene = "scene110";
+	}
+	return true;
+}
+
+// misc game functions
+void GameSpacePirates::_PlayErrorSound() {
+	_PlaySound(_skullSound);
+}
+
+void GameSpacePirates::_DisplayShotFiredImage() {
+	_DisplayShotFiredImage(&_mousePos);
+}
+
+void GameSpacePirates::_DisplayShotFiredImage(Common::Point *point) {
+	if (point->x >= _videoPosX && point->x <= (_videoPosX + _videoDecoder->getWidth()) && point->y >= _videoPosY && point->y <= (_videoPosY + _videoDecoder->getHeight())) {
+		uint16 targetX = point->x - _videoPosX - 4;
+		uint16 targetY = point->y - _videoPosY - 4;
+		AlgGraphics::drawImageCentered(_videoDecoder->getVideoFrame(), &_bulletholeIcon, targetX, targetY);
+	}
+}
+
+void GameSpacePirates::_DisplayShotLine(uint16 startX, uint16 startY, uint16 endX, uint16 endY) {
+	int16 currentX = startX;
+	int16 currentY = startY;
+	int16 deltaX = endX - startX;
+	int16 deltaY = endY - startY;
+	int16 stepX, stepY;
+	int16 steep, accum, error;
+	if (deltaX > 0)
+		stepX = 1;
+	else if (deltaX == 0)
+		stepX = 0;
+	else
+		stepX = -1;
+	if (deltaY > 0)
+		stepY = 1;
+	else if (deltaY == 0)
+		stepY = 0;
+	else
+		stepY = -1;
+	deltaX = abs(deltaX);
+	deltaY = abs(deltaY);
+	steep = (deltaX > deltaY) ? deltaX : deltaY;
+	for (int16 count = 0; count <= steep; count++) {
+		if (count % 10 == 0) {
+			Common::Point position = Common::Point(currentX, currentY);
+			_DisplayShotFiredImage(&position);
+		}
+		accum += deltaX;
+		error += deltaY;
+		if (accum > steep) {
+			accum -= steep;
+			currentX += stepX;
+		}
+		if (error > steep) {
+			error -= steep;
+			currentY += stepY;
+		}
+	}
+}
+
+void GameSpacePirates::_DisplayMultipleShotLines() {
+	_DisplayShotLine(0x73, 0x37, _mousePos.x, _mousePos.y);
+	_DisplayShotLine(0x20, 0x9B, _mousePos.x, _mousePos.y);
+	_DisplayShotLine(0x73, 0xFB, _mousePos.x, _mousePos.y);
+}
+
+void GameSpacePirates::_EnableVideoFadeIn() {
+	// TODO implement
+}
+
+uint16 GameSpacePirates::_SceneToNumber(Scene *scene) {
+	return atoi(scene->name.substr(5).c_str());
+}
+
+uint16 GameSpacePirates::_RandomUnusedScene(uint8 max) {
+	bool found = 0;
+	uint8 random;
```

Comment:
`random` can potentially be uninitialized, as the Windows builds show :)

```suggestion
	uint8 random = 0;
```

Reply:
Same, on it.

Reply:
Also should be fixed now, I think